kafka-admin/src/main/resources/graphql-schema/kafka-admin.graphql

```diff
@@ -1,11 +1,67 @@
-extend type Query {
-    topic(topicName: String): Topic
-    topicList(filter: String): [Topic]
+type TopicDescription {
+    #id: ID
+    name: String
+
+    """@transient"""
+    partitions: [Partitions]
+    config: TopicConfig
+}
+
+type Partitions {
+    partition: Int
+    replicas: [Replicas]
+}
+
+type Replicas {
+    id: String
+    inSync: Boolean
 }
 
-type Topic {
+type TopicOnlyName {
+    #id: ID
```

Comment:
In GraphQL ID's are sacred due to client side caching. If we get list of the topics and then operate on specific topics to get extra data client side will usually normalize those etc. 

Frameworks like Apollo etc. tend to have extra configuration to handle objects without id's etc. If we apply suggestion from above this would not be issue